lib/galaxy/util/__init__.py

```diff
@@ -848,12 +848,20 @@ def unicodify( value, encoding=DEFAULT_ENCODING, error='replace', default=None )
     """
     Returns a unicode string or None
     """
-
-    if isinstance( value, text_type ):
-        return value
+    if value is None:
+        return None
+    if isinstance(value, bool):
+        return text_type(value)
+    if isinstance( value, string_types):
+        if isinstance( value, text_type ):
+            return value
+        else:
+            return text_type( value, encoding, error )
     try:
-        return text_type( str( value ), encoding, error )
-    except:
+        return text_type( value, encoding, error )
+    except Exception as e:
+        log.debug("value %s could not be coerced to unicode" % value)
+        log.debug(e)
         return default
```

Comment:
I've updated the above code to have also the call to text_type() inside the try.